### ISSUE TITLE
Minimal GPU memory requirement

### ISSUE THREAD

**Issue by bryanbocao** [Jiamian-Wang/S2-transformer-HSI#1]
Thanks for sharing the code! What is the minimal GPU memory requirement? The paper mentions RTX 3090 which has 24 GB GPU memory. Can a 8GB memory card like RTX 2080 Super/1080 be sufficient for training and inference?

Thanks!

**Comment by Jiamian-Wang** [comment]
Hi, thanks for your interst in our recent work! To train the full model on the benchmark HSI training set, one needs 2 RTX3090 GPUs, with the batch size of 4. Training the proposed network structure upon a single GPU with 8Gb would be quite struggling. Here are several tips that will help you relieve the computational overhead: (1) reduce the the number of S^2-attn stages, (2) reduce the the number of S^2-attn blocks (3) reduce the number of embedding channels (4) reduce the batch size, etc.

**Comment by bryanbocao** [comment]
@Jiamian-Wang Thanks for the reply! Lambda GPU workstation also has similar settings but now they provide 2 RTX 4090. So it's trained on two 24GB GPUs.

I read the paper and it is well written. 

For practical training time estimation,
(1) May I know **how long does it take to train a K=4, L=6, T=6, M=8, batch_size=4 for 400 epochs with 2 RTX 4090 GPUs in terms of hours (or days)?**

I noticed that the pretrained model [model_epoch_255.pth](https://ufile.io/bmxw9qdn) has 224MB, **(2) what is the minimal GPU memory for the model?** My understand is that # stages, # blocks and # embedding channels affect GPU memory requirements for a model, while batch size determines the amount of measurements and masks saved in GPU memory.
**(3) Is a 8GB GPU sufficient just for inference/testing using a pretrained model (without training)?**
Thanks!
